Mobile workspace

mehmetSuzer opened the pull request:
The design of the mobile workspace pages and the navigations between them are implemented without providers. 
The models of @Zulalm are used in the implementation.
Once the providers are implemented, the pages will be updated accordingly.

mehmetSuzer commented:
### Some Additional Changes That I Made

1. Since there are many pages which cannot be accessed without logging in, I parametrized the PleaseLoginPage. It takes a message. Different messages can be displayed for different pages. 

2. Files for creating workspace pages are added, but they do not function yet.

3. PageWithAppBar takes a FloatingActionButton now. In default case, there is no FloatingActionButton .

4. I turned AccountSettingsAppBar into a parametrized widget named SimpleAppBar(message).

Zulalm commented on project/FrontEnd/collaborative_science_platform/lib/screens/workspaces_page/workspaces_page.dart:
Since there is not a web workspaces page, this file also should be updated accordingly.

Zulalm reviewed (changes requested):
The design of the page is looks good but I have some minor comments to improve UX. Bold black text style is not easy to read on a dark blue surface. Also, I think we should not use primary color on all clickable entries. We should use the primary color for the clickable objects that we want to get attention of the user. However, these are only design choices we may discuss them.

laylaylo commented on project/FrontEnd/collaborative_science_platform/lib/models/annotation_text.dart:
I realized this since I got the same import error for annotation text.
Please don't include this file as @omerfaunal moved it to widgets directory, instead change import path in models/annotation.dart

mehmetSuzer replied:
OK, I changed the path of the import.

mehmetSuzer commented on project/FrontEnd/collaborative_science_platform/lib/main.dart:
```diff
@@ -19,7 +19,7 @@ void main() {
 void configureApp() {
   if (kIsWeb) {
-    setUrlStrategy(PathUrlStrategy());
+    setUrlStrategy(const PathUrlStrategy());
```
It does not cause any problem in my Android emulator. Testing the code in Chrome might be the problem.

mehmetSuzer commented:
I changed the colors of the widgets, removed unnecessary files, modified the path of annotationText, changed the weight of the texts.

I also fixed an overflow bug in the graph mobile page.

Now, the user him/herself is automatically added to the list of contributors in the workspace.

omerfaunal reviewed (approved):
Changed file structure, and everything seems OK
